File: app/src/main/java/com/joxad/zikobot/app/core/module/soundcloud/model/SoundCloudTrack.java

package com.joxad.zikobot.app.core.module.soundcloud.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

import lombok.AllArgsConstructor;
import lombok.Data;

@Parcel
@Data
public class SoundCloudTrack {

    @SerializedName("kind")
    @Expose
    public String kind;
    @SerializedName("id")
    @Expose
    public Integer id;
    @SerializedName("created_at")
    @Expose
    public String createdAt;
    @SerializedName("user_id")
    @Expose
    public Integer userId;
    @SerializedName("duration")
    @Expose
    public Integer duration;
    @SerializedName("commentable")
    @Expose
    public Boolean commentable;
    @SerializedName("state")
    @Expose
    public String state;
    @SerializedName("original_content_size")
    @Expose
    public Integer originalContentSize;
    @SerializedName("sharing")
    @Expose
    public String sharing;
    @SerializedName("tag_list")
    @Expose
    public String tagList;
    @SerializedName("permalink")
    @Expose
    public String permalink;
    @SerializedName("streamable")
    @Expose
    public Boolean streamable;
    @SerializedName("embeddable_by")
    @Expose
    public String embeddableBy;
    @SerializedName("downloadable")
    @Expose
    public Boolean downloadable;
    @SerializedName("genre")
    @Expose
    public String genre;
    @SerializedName("title")
    @Expose
    public String title;
    @SerializedName("description")
    @Expose
    public String description;
    @SerializedName("label_name")
    @Expose
    public String labelName;
    @SerializedName("release")
    @Expose
    public String release;
    @SerializedName("track_type")
    @Expose
    public String trackType;
    @SerializedName("key_signature")
    @Expose
    public String keySignature;
    @SerializedName("isrc")
    @Expose
    public String isrc;

    @SerializedName("original_format")
    @Expose
    public String originalFormat;
    @SerializedName("license")
    @Expose
    public String license;
    @SerializedName("uri")
    @Expose
    public String uri;
    @SerializedName("user")
    @Expose
    public User user;
    @SerializedName("created_with")
    @Expose
    public CreatedWith createdWith;
    @SerializedName("permalink_url")
    @Expose
    public String permalinkUrl;
    @SerializedName("artwork_url")
    @Expose
    public String artworkUrl;
    @SerializedName("waveform_url")
    @Expose
    public String waveformUrl;
    @SerializedName("stream_url")
    @Expose
    public String streamUrl;
    @SerializedName("playback_count")
    @Expose
    public Integer playbackCount;
    @SerializedName("download_count")
    @Expose
    public Integer downloadCount;
    @SerializedName("favoritings_count")
    @Expose
    public Integer favoritingsCount;
    @SerializedName("comment_count")
    @Expose
    public Integer commentCount;
    @SerializedName("attachments_uri")
    @Expose
    public String attachmentsUri;
    @SerializedName("download_url")
    @Expose
    public String downloadUrl;

}

File: data/src/main/java/com/joxad/zikobot/data/module/soundcloud/model/SoundCloudTrack.kt
package com.joxad.zikobot.data.module.soundcloud.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


class SoundCloudTrack {

    @SerializedName("kind")
    @Expose
    var kind: String? = null
    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("created_at")
    @Expose
    var createdAt: String? = null
    @SerializedName("user_id")
    @Expose
    var userId: Int? = null
    @SerializedName("duration")
    @Expose
    var duration: Int? = null
    @SerializedName("commentable")
    @Expose
    var commentable: Boolean? = null
    @SerializedName("state")
    @Expose
    var state: String? = null
    @SerializedName("original_content_size")
    @Expose
    var originalContentSize: Int? = null
    @SerializedName("sharing")
    @Expose
    var sharing: String? = null
    @SerializedName("tag_list")
    @Expose
    var tagList: String? = null
    @SerializedName("permalink")
    @Expose
    var permalink: String? = null
    @SerializedName("streamable")
    @Expose
    var streamable: Boolean? = null
    @SerializedName("embeddable_by")
    @Expose
    var embeddableBy: String? = null
    @SerializedName("downloadable")
    @Expose
    var downloadable: Boolean? = null
    @SerializedName("genre")
    @Expose
    var genre: String? = null
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("label_name")
    @Expose
    var labelName: String? = null
    @SerializedName("release")
    @Expose
    var release: String? = null
    @SerializedName("track_type")
    @Expose
    var trackType: String? = null
    @SerializedName("key_signature")
    @Expose
    var keySignature: String? = null
    @SerializedName("isrc")
    @Expose
    var isrc: String? = null

    @SerializedName("original_format")
    @Expose
    var originalFormat: String? = null
    @SerializedName("license")
    @Expose
    var license: String? = null
    @SerializedName("uri")
    @Expose
    var uri: String? = null
    @SerializedName("user")
    @Expose
    var user: SoundCloudUser? = null
    @SerializedName("created_with")
    @Expose
    var createdWith: CreatedWith? = null
    @SerializedName("permalink_url")
    @Expose
    var permalinkUrl: String? = null
    @SerializedName("artwork_url")
    @Expose
    var artworkUrl: String? = null
    @SerializedName("waveform_url")
    @Expose
    var waveformUrl: String? = null
    @SerializedName("stream_url")
    @Expose
    var streamUrl: String? = null
    @SerializedName("playback_count")
    @Expose
    var playbackCount: Int? = null
    @SerializedName("download_count")
    @Expose
    var downloadCount: Int? = null
    @SerializedName("favoritings_count")
    @Expose
    var favoritingsCount: Int? = null
    @SerializedName("comment_count")
    @Expose
    var commentCount: Int? = null
    @SerializedName("attachments_uri")
    @Expose
    var attachmentsUri: String? = null
    @SerializedName("download_url")
    @Expose
    var downloadUrl: String? = null

}
